aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -336,6 +336,17 @@ void index_func_meta_impl(
               func, "_(): Number of indices (", numel, ") should be equal to source.size(dim): (",
               source.size(dim), "), for dim: ", dim);
 
+  if (source.dim() != 0) {
+    auto self_sizes = self.sizes().vec();
+    auto source_sizes = source.sizes().vec();
+    self_sizes.erase(self_sizes.begin() + dim);
+    source_sizes.erase(source_sizes.begin() + dim);
+    auto self_sizes_tensor = at::tensor(self_sizes, index.options());
+    auto source_sizes_tensor = at::tensor(source_sizes, index.options());
```

Comment:
Why creating a tensor just to call `.sizes()`? Pass the sizes directly.

Reply:
I tried this initially, but it wasn't working. ```test/test_legacy_vmap.py TestVmapAPI.test_fallback_masked_fill``` errors out when I try running it locally if passing the self_sizes and source_sizes directly (after erasure).

My assumption is that we need to call .vec() to erase the given dimension, and in order to convert back to the .sizes() representation, we must create tensors that we can call .sizes() on. This assumption might be faulty but I'm unsure how to transition from the vector representations back to IntArrayRefs is_expandable_to accepts otherwise.

Reply:
I tried converting the sizes into IntArrayRefs with a constructor, but this wasn't working well. Is there a better way to go from a vector representation to the one calling .sizes() on a tensor produces?

Reply:
if you just pass the vectors, they should be casted automatically to `IntArrayRef`s via the implicit constructor.